x/leverage/keeper/interest.go

```diff
@@ -133,11 +133,14 @@ func (k Keeper) AccrueAllInterest(ctx sdk.Context) error {
 			interestAccrued.TruncateInt(),
 		))
 
-		// calculate new reserves accrued for this denom
-		newReserves = newReserves.Add(sdk.NewCoin(
-			token.BaseDenom,
-			interestAccrued.Mul(token.ReserveFactor).TruncateInt(),
-		))
+		// if interest accrued on this denom is at least one base token
+		if interestAccrued.GT(sdk.OneDec()) {
```

Comment:
shouldn't be `GT 0`?

Reply:
```suggestion
		if interestAccrued.GT(sdk.ZeroDec()) {
```

Reply:
This decimal represents base tokens, so if it's `0.5` (i.e. `0.5uumee`) it must be treated as `0uumee`, not `1uumee` to prevent more reserves from being allocated than interest.

(in that case, uToken exchange rate would decrease which we should not permit)